lib/src/layout/scaffold.dart

```diff
@@ -1,117 +1,303 @@
-  final double sidebarBreakpoint;
+class _ScaffoldState extends State<Scaffold> {
+  final _sidebarScrollController = ScrollController();
+  final _minContentAreaWidth = 300.0;
+  ResizablePaneNotifier _valueNotifier = ResizablePaneNotifier({});
+  double _sidebarWidth = 0.0;
 
-  /// The color of the body/sidebar splitter
-  final Color? sidebarGripColor;
+  @override
+  void initState() {
+    super.initState();
+    _sidebarWidth = widget.sidebar?.minWidth ?? _sidebarWidth;
+    if (widget.sidebar?.builder != null)
+      _sidebarScrollController.addListener(() => setState(() {}));
+    WidgetsBinding.instance?.addPostFrameCallback((_) {
+      _valueNotifier.notify();
+    });
+  }
 
-  /// The width of the split between [body] and [sidebar].
-  ///
-  /// Defaults to 0.80, which seems to be the default in Apple's macOS apps
-  /// (I eyeballed this so it's not perfect but it's very close).
-  final double sidebarGripSize;
+  @override
+  void dispose() {
+    _sidebarScrollController.dispose();
+    super.dispose();
+  }
 
-  /// Determines where the split between [body] and [sidebar] occurs.
-  ///
-  /// If specified, it must be a value greater than 0.0 and less than 1.0.
-  ///
-  /// Defaults to `0.25`, which is 1/4 of the available space from the left.
-  final double splitOffset;
+  @override
+  void didUpdateWidget(covariant Scaffold old) {
+    super.didUpdateWidget(old);
+    WidgetsBinding.instance?.addPostFrameCallback((_) {
+      _valueNotifier.reset();
+      setState(() {
+        if (widget.sidebar == null)
+          _sidebarWidth = 0.0;
+        else {
+          if (widget.sidebar!.minWidth > _sidebarWidth)
+            _sidebarWidth = widget.sidebar!.minWidth;
+          if (widget.sidebar!.maxWidth! < _sidebarWidth)
+            _sidebarWidth = widget.sidebar!.maxWidth!;
+        }
+      });
+      _valueNotifier.notify();
+    });
+  }
 
   @override
+  // ignore: code-metrics
   Widget build(BuildContext context) {
     debugCheckHasMacosTheme(context);
+    assert(
+      widget.children.every((e) => e is ContentArea || e is ResizablePane),
+      'Scaffold children must either be ResizablePane or ContentArea',
+    );
+    assert(
+      widget.children.whereType<ContentArea>().length <= 1,
+      'Scaffold cannot have more than one ContentArea widget',
+    );
 
-    final style = context.macosTheme;
-    late Color bodyColor;
-    late Color sidebarColor;
-    late Color gripColor;
+    final theme = context.macosTheme;
+    late Color backgroundColor;
+    late Color sidebarBackgroundColor;
+    Color dividerColor = theme.dividerColor!;
 
-    if (style.brightness == Brightness.light) {
-      bodyColor = backgroundColor ?? CupertinoColors.systemBackground.color;
-      sidebarColor =
-          sidebarBackgroundColor ?? CupertinoColors.tertiarySystemBackground;
-      gripColor = sidebarGripColor ?? CupertinoColors.systemFill;
+    if (!theme.brightness!.isDark) {
```

Comment:
I would recommed to create a new local `CupertinoDynamicColor` to avoid boilerplate code. Just a recommendation, but it's good the way it is